src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstodeliusupdate/services/CourtRegisterSyncService.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.services
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsregisterstodeliusupdate.services.ProbationService.CourtFromProbationSystem
+
+@Service
+class CourtRegisterSyncService(
+  private val courtRegisterUpdateService: CourtRegisterUpdateService,
+  private val courtRegisterService: CourtRegisterService,
+  private val probationService: ProbationService
+) {
+
+  fun sync(): SyncStatistics {
+    return syncAllCourts(probationService.getAllCourts(), courtRegisterService.getAllActiveCourts())
+  }
+
+  private fun syncAllCourts(
+    probationCourts: List<CourtFromProbationSystem>,
+    courtRegisterCourts: List<CourtDto>
+  ): SyncStatistics {
+
+    // get all active / inactive courts from Probation (DELIUS)
+    val allCourtsHeldInProbation = probationCourts.map { ProbationService.CourtDataToSync(it) }.associateBy { it.code }
+
+    // get all the courts from the register
+    val allRegisteredCourts: MutableList<ProbationService.CourtDataToSync> = mutableListOf()
+    courtRegisterCourts.forEach {
+      allRegisteredCourts.add(ProbationService.CourtDataToSync(it))
+    }
+    val courtRegisterMap = allRegisteredCourts.associateBy { it.code }
+
+    val stats = SyncStatistics()
+    // matches
+    courtRegisterMap.filter { c -> allCourtsHeldInProbation[c.key] != null }
```

Comment:
You could maybe move each of these filters up into one line functions with nice names like `matches`, `new`, `missing`?  Just one of those cases where you could do the "remove the comments with self-documenting kind" type approach.